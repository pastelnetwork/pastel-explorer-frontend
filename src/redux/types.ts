import { ThunkAction, ThunkDispatch } from 'redux-thunk';
import { Action } from 'redux';
import { RootState } from '@redux/store';

export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;

export type AppThunkDispatch = ThunkDispatch<RootState, unknown, Action<string>>;
