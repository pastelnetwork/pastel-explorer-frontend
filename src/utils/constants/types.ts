import { LabelKeyObject } from 'react-csv/components/CommonPropTypes';
import { PeriodTypes, TGranularity } from '@utils/helpers/statisticsLib';
import { TStatisticsInfo } from '@utils/types/IStatistics';

export type TThemeColor = {
  name: string;
  backgroundColor: string;
  lineColor?: string;
  splitLineColor: string;
  color: string;
  stack?: string;
  smooth?: boolean;
};

export type TThemeInitOption = {
  theme?: TThemeColor | null;
  data?: number[][];
  dataX?: string[];
  dataY?: number[];
  dataY1?: number[];
  dataY2?: number[];
  chartName: string;
  minY: number;
  maxY: number;
};

export type TLineChartProps = {
  chartName: string;
  period?: PeriodTypes;
  dataX?: string[];
  dataY?: number[];
  dataY1?: number[];
  dataY2?: number[];
  title?: string;
  granularity?: TGranularity;
  info: TStatisticsInfo;
  granularities?: TGranularity[];
  offset: number;
  periods?: PeriodTypes[];
  handleGranularityFilterChange?: (_granularity: TGranularity) => void;
  handleBgColorChange: (_color: string) => void;
  handlePeriodFilterChange?: (_period: PeriodTypes) => void;
};

export type TCsvHeaderType = {
  [key: string]: LabelKeyObject[];
};

export type TScatterChartProps = {
  chartName: string;
  data: number[][];
  dataX: string[];
  title?: string;
  period?: PeriodTypes;
  info: TStatisticsInfo;
  offset: number;
  periods: PeriodTypes[];
  handleBgColorChange: (_color: string) => void;
  handlePeriodFilterChange?: (_period: PeriodTypes) => void;
  handleGranularityFilterChange?: (_granularity: TGranularity) => void;
};
