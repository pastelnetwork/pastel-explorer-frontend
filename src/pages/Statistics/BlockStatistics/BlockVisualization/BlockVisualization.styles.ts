import styled from 'styled-components';

import { lighten } from '@material-ui/core';

import themeVariant from '@theme/variants';

export const BlockContainer = styled.div`
  display: inline-flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  position: relative;
  color: ${themeVariant.custom.white};
  text-align: center;
  width: 125px;
  height: 125px;
  cursor: pointer;
  background: repeating-linear-gradient(
    rgb(45, 51, 72),
    rgb(45, 51, 72) 0.175375%,
    rgb(147, 57, 244) 0.175375%,
    rgb(16, 95, 176) 100%
  );
  &.block-unconfirmed {
    background: repeating-linear-gradient(
      to right,
      rgb(85, 75, 69),
      rgb(85, 75, 69) 0.7863%,
      rgb(93, 125, 1) 0.7863%,
      rgb(93, 125, 1) 14.6762%,
      rgb(93, 125, 1) 14.9597%,
      rgb(109, 125, 4) 29.5583%,
      rgb(109, 125, 4) 29.1331%,
      rgb(140, 125, 9) 43.4482%,
      rgb(140, 125, 9) 43.3065%,
      rgb(166, 125, 14) 57.3381%,
      rgb(166, 125, 14) 57.4798%,
      rgb(166, 125, 14) 71.228%,
      rgb(166, 125, 14) 71.6532%,
      rgb(170, 125, 15) 86.1101%,
      rgb(170, 125, 15) 85.8266%,
      rgb(178, 125, 16) 100%
    );
  }
  // background: repeating-linear-gradient(
  //   ${themeVariant.palette.primary.main},
  //   ${themeVariant.palette.primary.main} 0.018575%,
  //   ${themeVariant.palette.secondary.main} 0.018575%,
  //   ${themeVariant.palette.primary.main} 100%
  // );
  transition: transform 0.2s ease-in;

  &::before {
    content: '';
    width: 20px;
    height: 125px;
    position: absolute;
    top: -12px;
    left: -20px;
    background-color: ${themeVariant.sidebar.background};
    transform: skewY(50deg);
    transform-origin: top;
  }

  &::after {
    content: '';
    width: 125px;
    height: 24px;
    position: absolute;
    top: -24px;
    left: -20px;
    background-color: ${lighten(themeVariant.sidebar.background, 0.2)};
    transform: skew(40deg);
    transform-origin: top;
  }

  &:hover {
    transform: translateY(-2px);
  }
`;
