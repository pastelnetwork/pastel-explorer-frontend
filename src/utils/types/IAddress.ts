export interface IAddressData {
  amount: number;
  timestamp: number;
  transactionHash: string;
  direction: string;
}

export interface IAddress {
  address: string;
  incomingSum: number;
  outgoingSum: number;
  data: Array<IAddressData>;
}
