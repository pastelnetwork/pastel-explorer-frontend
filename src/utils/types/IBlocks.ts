export interface IBlockTransaction {
  id: string;
  recipientCount: number;
  totalAmount: number;
}

export interface IBlock {
  confirmations: number;
  difficulty: string;
  height: number;
  id: string;
  merkleRoot: string;
  nextBlockHash: string;
  nonce: string;
  previousBlockHash: string;
  size: number;
  solution: string;
  timestamp: number;
  transactionCount: number;
  transactions: Array<IBlockTransaction>;
}

export interface IRawBlock {
  anchor: string;
  bits: string;
  chainwork: string;
  confirmations: number;
  difficulty: string;
  finalsaplingroot: string;
  hash: string;
  height: number;
  merkleroot: string;
  nonce: string;
  previousblockhash: string;
  size: number;
  solution: string;
  time: number;
  transactions: Array<IBlockTransaction>;
  tx: string[];
}
