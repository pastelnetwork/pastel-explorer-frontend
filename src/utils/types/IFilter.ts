export type TFilter = {
  name: string;
  value: string;
};
