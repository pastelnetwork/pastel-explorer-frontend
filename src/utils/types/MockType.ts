import { TFunction } from 'i18next';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type MyMockType = jest.Mock<any, any, any> &
  TFunction<['translation', ...string[]], undefined>;
