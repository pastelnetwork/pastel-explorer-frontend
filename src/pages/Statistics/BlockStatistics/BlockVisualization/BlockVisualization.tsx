import { Typography } from '@material-ui/core';

import * as Styles from './BlockVisualization.styles';

type BlockElementType = string | number | JSX.Element;

interface BlockVisualizationProps {
  height: BlockElementType;
  size: BlockElementType;
  transactionCount: BlockElementType;
  minutesAgo: BlockElementType;
  clickHandler?: () => void;
  className?: string;
}

const BlockVisualization: React.FC<BlockVisualizationProps> = ({
  height,
  size,
  transactionCount,
  minutesAgo,
  className = '',
  clickHandler,
}) => {
  return (
    <Styles.BlockContainer className={className} onClick={clickHandler}>
      <Typography variant="caption">Block #:</Typography>
      <Typography variant="h3">{height}</Typography>
      <Typography variant="h4">{size}</Typography>
      <Typography variant="caption">{transactionCount}</Typography>
      <Typography variant="caption">{minutesAgo}</Typography>
    </Styles.BlockContainer>
  );
};

export default BlockVisualization;
