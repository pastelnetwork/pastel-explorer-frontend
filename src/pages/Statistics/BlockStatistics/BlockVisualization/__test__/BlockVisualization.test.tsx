import { shallow } from 'enzyme';

import 'jest-styled-components';

import { MyMockType } from '@utils/types/MockType';
import i18next from '../../../../../utils/helpers/i18n';
import BlockVisualization from '../BlockVisualization';

jest.mock('i18next-http-backend');
jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () =>
          new Promise(() => {
            // noop
          }),
      },
    };
  },
  initReactI18next: {
    type: '3rdParty',
    init: () => {
      // noop
    },
  },
}));
i18next.t = jest.fn().mockImplementation((...arg) => {
  return arg[0];
}) as MyMockType;

describe('pages/Statistics/BlockStatistics/BlockVisualization', () => {
  const data = {
    id: '245684',
    transactionCount: '3',
    height: '245684',
    size: '2.5kB',
    minutesAgo: '3 minutes',
    ticketsCount: '0',
  };
  const wrapper = shallow(<BlockVisualization {...data} />);

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
