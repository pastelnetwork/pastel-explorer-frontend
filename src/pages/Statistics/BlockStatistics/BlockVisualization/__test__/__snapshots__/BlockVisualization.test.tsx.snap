// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Statistics/BlockStatistics/BlockVisualization renders correctly 1`] = `ShallowWrapper {}`;
